### PR TITLE
`yoshi-server` bug fix: return of different types

### LINKED ISSUE #1540: Support returning undefined from server functions
### Summary

Server functions can often be used for side effects and forcing a valid JSON return type (like an empty object or null) isn't very convenient.

### PR DESCRIPTION
**Pull request by yanivefraim**
fixes #1540 

Returning a string from a route / not returning a value causes `yoshi-server` to crash.

TODO: add a base fixture for `yoshi-server` tests, for code reuse.

### CODE REVIEW

File: packages/yoshi-server/src/server.ts [hunk 1]
```diff
@@ -93,7 +93,16 @@ export default class Server {
             params,
           };
 
-          return send(res, 200, await chunk.call(fnThis));
+          const result = await chunk.call(fnThis);
+          if (result) {
+            if (res.headersSent) {
```
Comment (ronami) [inline packages/yoshi-server/src/server.ts]: I think we should add a comment to explain why this is needed (it's not immediately obvious).

File: test/typescript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
+      const title = await page.$eval('h1', elm => elm.innerText);
+      expect(title).toBe('hello Yaniv');
+    });
+  });
+
+  it('have only one header', async () => {
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
```
Comment (ronami) [inline test/typescript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js]: we can use `scripts.url`

File: test/typescript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+const scripts = Scripts.setupProjectFromTemplate({
+  templateDir: __dirname,
+  projectType: Scripts.projectType.TS,
+});
+
+describe.each(['prod', 'dev'])('send a value directly [%s]', mode => {
+  it('integration', async () => {
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
```
Comment (ronami) [inline test/typescript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js]: same as bellow

File: test/typescript/features/yoshi-server/return-values/return-a-string/return-a-string.test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+const scripts = Scripts.setupProjectFromTemplate({
+  templateDir: __dirname,
+  projectType: Scripts.projectType.TS,
+});
+
+describe.each(['prod', 'dev'])('return a string [%s]', mode => {
+  it('run tests', async () => {
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
```
Comment (ronami) [inline test/typescript/features/yoshi-server/return-values/return-a-string/return-a-string.test.js]: `scripts.url`

File: test/javascript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
+      const title = await page.$eval('h1', elm => elm.innerText);
+      expect(title).toBe('hello Yaniv');
+    });
+  });
+
+  it('have only one header', async () => {
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
```
Comment (ronami) [inline test/javascript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js]: `scripts.url`

File: test/javascript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+const scripts = Scripts.setupProjectFromTemplate({
+  templateDir: __dirname,
+  projectType: Scripts.projectType.JS,
+});
+
+describe.each(['prod', 'dev'])('send a value directly [%s]', mode => {
+  it('integration', async () => {
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
```
Comment (ronami) [inline test/javascript/features/yoshi-server/return-values/send-a-response-directly/send-a-response-directly.test.js]: `scripts.url`

File: test/javascript/features/yoshi-server/return-values/return-a-string/return-a-string.test.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+const scripts = Scripts.setupProjectFromTemplate({
+  templateDir: __dirname,
+  projectType: Scripts.projectType.JS,
+});
+
+describe.each(['prod', 'dev'])('return a string [%s]', mode => {
+  it('run tests', async () => {
+    await scripts[mode](async () => {
+      await page.goto(`http://localhost:3000/app`);
```
Comment (ronami) [inline test/javascript/features/yoshi-server/return-values/return-a-string/return-a-string.test.js]: `scripts.url`

File: test/javascript/features/yoshi-server/return-values/return-a-string/environment.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    .val('scripts_domain', 'static.parastorage.com')
+    .emit();
+};
+
+// start the server as an embedded app
+module.exports.bootstrapServer = () => {
+  return testkit.app(require.resolve('yoshi-server/bootstrap'), {
+    env: process.env,
+  });
+};
```
Comment (ronami) [inline test/javascript/features/yoshi-server/return-values/return-a-string/environment.js]: Some of these files are repeated quite a bit. I think we should refactor this (possibly in a future PR) into a fixture. Wdyt?

File: packages/yoshi-server/src/server.ts (hunk 1 above)
Comment (yanivefraim) [reply packages/yoshi-server/src/server.ts]: added

**Pull request merged**